framework/src/base/FEProblem.C

```diff
@@ -3561,6 +3564,9 @@ FEProblem::computeJacobian(NonlinearImplicitSystem & sys, const NumericVector<Nu
     execTransfers(EXEC_NONLINEAR);
     execMultiApps(EXEC_NONLINEAR);
 
+    for (unsigned int tid = 0; tid < n_threads; tid++)
+      reinitScalars(tid);
```

Comment:
Is this necessary here? We already reinitialize scalars in `NonlinearSystem::computeJacobianInternal`, `NonlinearSystem::computeResidualInternal`, and other places. Is it to get the scalars initialized for the UserObjects? It probably doesn't add a noticeable overheat, but it adds it for everyone and each residual computation, so we should be careful here.